rotkehlchen/chain/ethereum/manager.py

```diff
@@ -1198,3 +1210,250 @@ def get_basic_contract_info(self, address: ChecksumEthAddress) -> Dict[str, Any]
 
         self.contract_info_cache[address] = info
         return info
+
+    def update_curve_pools_cache(self) -> None:
+        # Delete current cache. Need to do this in case curve removes some pools
+        GlobalDBHandler().delete_requests_cache(cache_type=RequestsCacheType.CURVE_POOL_ADDR)
+        GlobalDBHandler().delete_requests_cache(cache_type=RequestsCacheType.CURVE_POOL_TOKENS)
+        # write new values to the cache
+        self.update_cure_official_pools_cache()
+        self.update_curve_metapools_cache()
+
+    def update_cure_official_pools_cache(self) -> None:
```

Comment:
Add a docstring to this function and the erros that it can raise

Reply:
Also this function is big and hard to follow. Splitting it in smaller functions would help with readibility